File: INFORMÁTICA/PROGRAMACIÓN Y DISEÑO ORIENTADO A OBJETOS/PRÁCTICAS/PROYECTOS/EXÁMENES/examen_22-23_extraordinaria/deepspace/src/deepspace/TestP1.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;
import java.util.ArrayList;

//Programa principal
public class TestP1 {
    
    public static void main(String[] args){
        
        //WeaponType
        WeaponType weapon_t1 =WeaponType.LASER, weapon_t2=WeaponType.MISSILE, weapon_t3=WeaponType.PLASMA;
        
        System.out.println("\nWeaponType attributes:"
                +" \nPower:" 
                + "\nLASER= "   + weapon_t1.getPower()
                + "\nMISSILE= " + weapon_t2.getPower()
                + "\nPLASMA= "  + weapon_t3.getPower());
        
        
        //Loot
        int nSupplies=1;
        int nWeapons=2;
        int nShields=3;
        int nHangars=4;
        int nMedals=5;
        Loot loot =new Loot(nSupplies,nWeapons,nShields,nHangars,nMedals);
        System.out.println("\nLoot attributes:"
                + "\nnSupplies="+loot.getNSupplies()
                + "\nnWeapons=" +loot.getNWeapons()
                + "\nnShields=" +loot.getNShields()
                + "\nnHangars=" +loot.getNHangars()
                + "\nnMedals="  +loot.getNMedals());

        
        //Supplies Package
        float ammoPower=1.0f;
        float fuelUnits=2.0f;
        float shieldPower=3.0f;
        SuppliesPackage pkg= new SuppliesPackage(ammoPower,fuelUnits,shieldPower);
        System.out.println("\nSupplies Package attributes:"
                + "\nammoPower="   +pkg.getAmmoPower()
                + "\nfuelUnits="   +pkg.getFuelUnits()
                + "\nshieldPower=" +pkg.getShieldPower());
     
        
        //Shield Booster
        String shield_name = "The Shield";
        float boost=1.0f;
        int shield_uses=2;
        ShieldBooster shield =new ShieldBooster(shield_name,boost,shield_uses);
        System.out.println("\nShield Booster attributes:"
                + "\nboost=" +shield.getBoost()
                + "\nuses="  +shield.getUses());
        
        
        //Weapon
        String weapon_name="The Weapon";
        int weapon_uses=2;
        Weapon weapon = new Weapon(weapon_name,weapon_t1,weapon_uses);
        System.out.println("\nWeapon attributes:"
                + "\nweapon_type=" +weapon.getType()
                + "\nweapon_uses=" +weapon.getUses());
        
        
        //Dice
        Dice dice = new Dice();
        final int REPS=100, NPLAYERS=20;
        float speed = 0.5f;
        
        int iwnh_counter0=0, iwnh_counter1=0;
        int iwnw_counter1=0, iwnw_counter2=0, iwnw_counter3=0;
        int iwns_counter0=0, iwns_counter1=0;
        
        ArrayList<Integer> ws_counter = new ArrayList<Integer>();
        for (int i=0; i<NPLAYERS;i++) 
            ws_counter.add(i,0);
        
        
        int fs_counter0=0, fs_counter1=0;
        int ssm_counter0=0, ssm_counter1=0;
        
        for (int i=0; i<REPS; i++){
            
            //initWithNHangars
            int nhangars=dice.initWithNHangars();
            if (nhangars==0) iwnh_counter0++;
            else iwnh_counter1++;
            
            //initWithNWeapons
            int nweapons = dice.initWithNWeapons();
            if (nweapons==1) iwnw_counter1++;
            else if (nweapons==2) iwnw_counter2++;
            else iwnw_counter3++;
            
            //initWithNShields
            int nshields=dice.initWithNShields();
            if (nshields==0) iwns_counter0++;
            else iwns_counter1++;
            
            //whoStarts
            int player=dice.whoStarts(NPLAYERS);
            int new_value = (ws_counter.get(player))+1;
            ws_counter.set(player,new_value);
            
            //firstShot
            GameCharacter shot= dice.firstShot();
            if (shot==GameCharacter.SPACESTATION) fs_counter0++;
            else fs_counter1++;
            
            //spaceStationMoves
            boolean move= dice.spaceStationMoves(speed);
            if (move) ssm_counter0++;
            else ssm_counter1++;
        }
        
        System.out.println("\nNumber of hangars:\n0: "+ iwnh_counter0 + "\n1: "+ iwnh_counter1
                + "\nNumber of weapons:\n1:" + iwnw_counter1 + "\n2:" + iwnw_counter2 + "\n3:" + iwnw_counter3
                + "\nNumber of shields:\n0:" + iwns_counter0 + "\n1:" + iwnw_counter1);
        
        for (int i=0; i<ws_counter.size();i++) System.out.println("\nPlayer " + i +" starts: " + ws_counter.get(i) + " times.");
        
        System.out.println("\nFirst Shot:\nSpace Station: " + fs_counter0  + "\nEnemy Starship: " + fs_counter1
                +"\n\nSpace Station Moves:\nEscapes: "   + ssm_counter0 + "\nHits: "              + ssm_counter1
                +"\n\n");
        
        }
}

File: INFORMÁTICA/PROGRAMACIÓN Y DISEÑO ORIENTADO A OBJETOS/PRÁCTICAS/PROYECTOS/deepspace_3/deepspace/deespace_rb/TestP1.rb
#encoding: UTF-8
require_relative 'CombatResult'
require_relative 'GameCharacter'
require_relative 'WeaponType'
require_relative 'Loot'
require_relative 'SuppliesPackage'
require_relative 'ShieldBooster'
require_relative 'Weapon'
require_relative 'Dice'


module Deepspace
    class TestP1

        NPLAYERS=20
        REPS=100

        def main

            #CombatResult
            puts "\nCombatResult attributes:"

            puts CombatResult::ENEMYWINS
            puts CombatResult::NOCOMBAT
            puts CombatResult::STATIONESCAPES
            puts CombatResult::STATIONWINS


            #GameCharacter
            "\nGameCharacter attributes:"

            puts GameCharacter::ENEMYSTARSHIP
            puts GameCharacter::SPACESTATION
            

            #WeaponType
        
            puts "\nWeaponType attributes:\nPower:" 
            puts "LASER= "   + WeaponType::LASER.power.to_s
            puts "MISSILE= " + WeaponType::MISSILE.power.to_s
            puts "PLASMA= "  + WeaponType::PLASMA.power.to_s

            #Loot
            nSupplies=1
            nWeapons=2
            nShields=3
            nHangars=4
            nMedals=5

            loot =Loot.new(nSupplies,nWeapons,nShields,nHangars,nMedals)
            puts loot.to_s

            #Supplies Package
            ammoPower=1.0
            fuelUnits=2.0
            shieldPower=3.0
            pkg= SuppliesPackage.new(ammoPower,fuelUnits,shieldPower)
            puts pkg.to_s

            #Shield Booster
            shield_name = "The Shield"
            boost=1.0
            shield_uses=2
            shield = ShieldBooster.new(shield_name,boost,shield_uses)
            puts shield.to_s
            
            
            #Weapon
            weapon_name="The Weapon"
            weapon_uses=2
            weapon = Weapon.new(weapon_name,WeaponType::LASER,weapon_uses)
            puts weapon.to_s

            #Dice

            dice = Dice.new

            iwnh_counter0=0
            iwnh_counter1=0
            iwnw_counter1=0
            iwnw_counter2=0 
            iwnw_counter3=0
            iwns_counter0=0 
            iwns_counter1=0

            ws_counter = Array.new(NPLAYERS, 0)

            fs_counter0=0
            fs_counter1=0
            ssm_counter0=0
            ssm_counter1=0
            speed = 0.5

            for i in 1..REPS do

                #initWithNHangars
                nhangars = dice.initWithNHangars

                if nhangars == 0
                    iwnh_counter0 += 1
                else
                    iwnh_counter1 += 1
                
                end


                #initWithNWeapons
                nweapons = dice.initWithNWeapons

                if nweapons == 1
                    iwnw_counter1 += 1
                elsif weapon == 3
                        iwnh_counter2 += 1
                
                else
                    iwnw_counter3 += 1
                end

                #initWithNShields
                nshields = dice.initWithNShields

                if nshields == 0
                    iwns_counter0 += 1
                else
                    iwns_counter1 += 1
                
                end

                #whoStarts
                player = dice.whoStarts(NPLAYERS)

                ws_counter[player] += 1
                
                #firstShot
                shot= dice.firstShot
                if shot==GameCharacter::SPACESTATION
                     fs_counter0 += 1
                else 
                    fs_counter1 += 1
                
                end

                #spaceStationMoves
                move= dice.spaceStationMoves(speed)
                if move 
                    ssm_counter0 += 1
                else 
                    ssm_counter1 += 1

                end

            end

            puts "\nNumber of hangars:\n0: "+ iwnh_counter0.to_s + "\n1: "+ iwnh_counter1.to_s
            puts "\nNumber of weapons:\n1:" + iwnw_counter1.to_s + "\n2:" + iwnw_counter2.to_s + "\n3:" + iwnw_counter3.to_s
            puts "\nNumber of shields:\n0:" + iwns_counter0.to_s + "\n1:" + iwnw_counter1.to_s
        
            for i in 1..NPLAYERS do
                puts "\nPlayer " + i.to_s + " starts: " + ws_counter[i-1].to_s + " times."
            end
        
            puts "\nFirst Shot:\nSpace Station: "      + fs_counter0.to_s  + "\nEnemy Starship: " + fs_counter1.to_s
            puts "\n\nSpace Station Moves:\nEscapes: " + ssm_counter0.to_s + "\nHits: "           + ssm_counter1.to_s
            puts "\n\n"

        end

    end

    TestP1.new.main

end

